erpnext/stock/report/stock_balance/stock_balance.py

```diff
@@ -143,15 +141,24 @@ def get_item_warehouse_map(self):
 		item_warehouse_map = {}
 		self.opening_vouchers = self.get_opening_vouchers()
 
-		for entry in self.sle_entries:
-			group_by_key = self.get_group_by_key(entry)
-			if group_by_key not in item_warehouse_map:
-				self.initialize_data(item_warehouse_map, group_by_key, entry)
+		with frappe.db.unbuffered_cursor():
+			if self.filters.get("show_stock_ageing_data"):
+				self.sle_entries = self.sle_query.run(as_dict=True)
+			else:
+				self.sle_entries = self.sle_query.run(as_dict=True, as_iterator=True)
+
+			for entry in self.sle_entries:
+				group_by_key = self.get_group_by_key(entry)
+				if group_by_key not in item_warehouse_map:
+					self.initialize_data(item_warehouse_map, group_by_key, entry)
+
+				self.prepare_item_warehouse_map(item_warehouse_map, entry, group_by_key)
 
-			self.prepare_item_warehouse_map(item_warehouse_map, entry, group_by_key)
+				if self.opening_data.get(group_by_key):
+					del self.opening_data[group_by_key]
 
-			if self.opening_data.get(group_by_key):
-				del self.opening_data[group_by_key]
+		if not self.filters.get("show_stock_ageing_data"):
```

Comment:
We can delete it in all cases? This is done just to avoid someone in future using this emptied iterator.

Reply:
In case of stock ageing we are using the sle entries therefore not deleted if show_stock_ageing_data is enabled

Reply:
Fixed